src/main/java/net/minecraftforge/eventbus/LMFListenerFactory.java

```diff
@@ -0,0 +1,61 @@
+package net.minecraftforge.eventbus;
+
+import net.minecraftforge.eventbus.api.Event;
+import net.minecraftforge.eventbus.api.IEventListener;
+
+import java.lang.invoke.LambdaMetafactory;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+
+public class LMFListenerFactory implements IEventListenerFactory {
+    /**
+     * We would prefer to use privateLookupIn(), but that does not set the MODULE bit,
+     * which LMF requires for class definition. So we have to use the IMPL_LOOKUP.
+     */
+    private static final MethodHandles.Lookup IMPL_LOOKUP;
+
+    static {
+        try {
+            var hackfield = MethodHandles.Lookup.class.getDeclaredField("IMPL_LOOKUP");
+            hackfield.setAccessible(true);
+            IMPL_LOOKUP = (MethodHandles.Lookup) hackfield.get(null);
+        } catch (Exception e) {
+            // TODO: if this fails I suppose we could fall back to a slower methodhandle invoke
+            throw new RuntimeException("Failed to access IMPL_LOOKUP...");
+        }
+    }
+
+    private static final MethodType LISTENER_INVOKE = MethodType.methodType(void.class, Event.class);
+
+    @Override
+    public IEventListener create(Method callback, Object target) {
```

Comment:
We need to double check that the resulting class and method names:
1. make sense in error messages
2. make sense in profiles